src/sentry/api/event_search.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+
+from collections import namedtuple
+from parsimonious.grammar import Grammar, NodeVisitor
+
+
+event_search_grammar = Grammar(r"""
+
+search          = filter+
+filter          = space? search_key sep search_value space?
+search_key      = ~r"[a-z]*\.?[a-z]*"
+search_value    = ~r"\S*"
+
+sep     = ":"
+space   = ~r"\s"
+"""
+                               )
+
+
+FIELD_LOOKUP = {
+    'user.id': {
+        'snuba_name': 'user_id',
+        'type': 'string',
+    },
+    'user.email': {
+        'snuba_name': 'email',
+        'type': 'string',
+    },
+    'release': {
+        'snuba_name': 'sentry:release',
+        'type': 'string',
+    }
+}
+
+
+class InvalidSearchQuery(Exception):
+    pass
+
+
+class SearchFilter(namedtuple('SearchFilter', 'key value')):
+    pass
+
+
+class SearchKey(namedtuple('SearchKey', 'name')):
+
+    @property
+    def snuba_name(self):
+        return FIELD_LOOKUP[self.name]['snuba_name']
+
+
+class SearchValue(namedtuple('SearchValue', 'raw_value type')):
+
+    @property
+    def parsed_value(self):
+        # TODO(jess): support parsing timestamps
```

Comment:
noob question: Would you do that through more grammar types?

Reply:
so i was going to talk to matt about this. i think you _could_ but i think we need to figure out if we want to attempt to guess types based on the format of the item (which we could do based on the grammar) or if we want to be more explicit and say these keys only are timestamps. does that make sense? i'm not totally sure yet which way would be better.

Reply:
Adding more constraints on the types of values should be doable. Linking those value types to specific key values may be harder, but might be doable. I've only done basic things with PEG parsers in the past.

Reply:
updated to handle in the grammar if you want to take a look